bin/oneoff/data_fix/fix_invalid_users

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env ruby
+
+require_relative '../../../dashboard/config/environment'
+
+# This script fixes invalid rows in the user table that were added to the database
+# as a result of a bug in https://github.com/code-dot-org/code-dot-org/pull/38026.
+# Briefly, there was a bug in that PR that caused user accounts created using the
+# join section endpoint (FollowersController#student_register) to be incomplete
+# or invalid (i.e. does not pass rails User model validation). The bug was present
+# only in v2020-12-02.0 which was in production from about 12/2/20 2:00pm PST to
+# 12/3/20 noon PST.
+#
+# The rows added to the user table were invalid in one of two ways:
+# 1. If the user passed valid data to the join section endpoint, the inserted row
+#    was valid except that the username field was missing. We fix these by
+#    calling generate_username to generate the username.
+# 2. If the user passed invalid data to the join section endpoint, the inserted row
+#    would contain the invalid data as well as have a missing provider field. We fix
+#    these row by (soft) deleting them.  In many cases, students were not able to
+#    access these accounts. For a small number of cases (n=25), students have
+#    accessed these accounts. For the students who successfully joined a section
+#    (n=9), we will reach out to the teacher to explain the situation.
```

Comment:
rather than "have a missing provider field" it might be more accurate to say "the inserted row would contain the invalid data, and would not successfully execute any post-save hooks on the user model, such as migration to multi-auth"